### PR TITLE
DOC: "feature.peak_local_max" : explanation of multiple same-intensity peaks returned by the function; added details on `exclude_border` parameter 

### LINKED ISSUE #2407: Better documentation for `peak_local_max`s `exclude_border` boolean variant
## Description

The documentation of [`peak_local_max`](http://scikit-image.org/docs/dev/api/skimage.feature.html#skimage.feature.peak_local_max) says nothing about the boolean variant of the `exclude_border` parameter, although it defaults to `True`. So it's unclear from the documentation what that is supposed to mean.

By looking at the corresponding [code](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/peak.py#L115), one can see that `min_distance` is used for `exclude_border` when it is passed `True`.

**First improvement:** Update the corresponding documentation to make this invariant clear.

Additionally, I think the default value of `True` might lead to subtle problems, that people might not be aware of. For instance, I wasn't aware of this fact until now and just found it by visual inspection. My mental model of NMS doesn't seem to fit the default implementation of `peak_local_max`, which I find problematic.

**Second improvement:** Don't use `True` as default for `exclude_border`, instead use `0`.

### PR DESCRIPTION
**Pull request by shcrela**
## Description
- Explained the fact of the function returning all off the same intensity peaks from the region defined with min_distance.
- Added the details on the `exclude_border` parameter behavior and the bool vs. int parameter values.

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
closes #2407

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/feature/peak.py [hunk 1]
```diff
@@ -51,9 +52,12 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
         the minimum intensity of the image.
     threshold_rel : float, optional
         Minimum intensity of peaks, calculated as `max(image) * threshold_rel`.
-    exclude_border : int, optional
-        If nonzero, `exclude_border` excludes peaks from
+    exclude_border : int or bool, optional
+        If nonzero int, `exclude_border` excludes peaks from
         within `exclude_border`-pixels of the border of the image.
+	If True, takes the `min_distance` parametar as value.
+	If zero or False, the peak is identified regardless of its'
```
Comment (emmanuelle) [inline skimage/feature/peak.py]: I would use plural instead "peaks are identified regardless of their distance from the border".

File: skimage/feature/peak.py [hunk 2]
```diff
@@ -51,9 +52,12 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
         the minimum intensity of the image.
     threshold_rel : float, optional
         Minimum intensity of peaks, calculated as `max(image) * threshold_rel`.
-    exclude_border : int, optional
-        If nonzero, `exclude_border` excludes peaks from
+    exclude_border : int or bool, optional
+        If nonzero int, `exclude_border` excludes peaks from
         within `exclude_border`-pixels of the border of the image.
+	If True, takes the `min_distance` parametar as value.
```
Comment (sciunto) [inline skimage/feature/peak.py]:
```suggestion
 	If True, takes the `min_distance` parameter as value.
```

File: skimage/feature/peak.py [hunk 3]
```diff
@@ -51,9 +52,12 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
...
+	If zero or False, peaks are identified regardless of their
+	distance from the border
```
Comment (sciunto) [inline skimage/feature/peak.py]:
```suggestion
 	distance from the border.
```

**Comment by emmanuelle** [comment]
Merging, thanks!

File: skimage/feature/peak.py [hunk 4]
```diff
@@ -51,9 +52,12 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
...
     threshold_rel : float, optional
         Minimum intensity of peaks, calculated as `max(image) * threshold_rel`.
-    exclude_border : int, optional
-        If nonzero, `exclude_border` excludes peaks from
+    exclude_border : int or bool, optional
+        If nonzero int, `exclude_border` excludes peaks from
         within `exclude_border`-pixels of the border of the image.
+	If True, takes the `min_distance` parameter as value.
+	If zero or False, peaks are identified regardless of their
+	distance from the border.
```
Comment (soupault) [inline skimage/feature/peak.py]: These tabs should be replaced by whitespaces.

**Pull request merged**